civihr_employee_portal/src/View/ViewsResultDiff.php

```diff
@@ -0,0 +1,284 @@
+      if ($diff) {
+        $dataDiff[$index] = $diff;
+      }
+    }
+    return $dataDiff;
+  }
+
+  /**
+   * Obtain $this->changes comparing $this->oldData with $this->newData
+   * @param  array $oldData
+   *     previous result from view
+   * @param  array $newData
+   *     a recent result from view
+   * @param  array $changes
+   *     the difference between $oldData and $newData
+   *
+   * @return stdClass $this
+   */
+  private function compareDataRecursive(&$oldData, &$newData, &$changes) {
+    static $i=-1;
+    $changes = !$i++ ? [] : $changes;
+
+    foreach ($oldData as $key => $dummyVar) {
+      // for groups we apply this function recursively
+      if (isset($oldData[$key]['group'])) {
+        $changes[] = ['group' => []];
+        $lastIndexOfChanges = count($changes) - 1;
+        $this->compareDataRecursive(
+          $oldData[$key]['group'],
+          $newData[$key]['group'],
+          $changes[$lastIndexOfChanges]['group']
+        );
+        // if there are no changes then this group must be deleted
+        if (!$changes[$lastIndexOfChanges]['group']) {
+          unset($changes[$lastIndexOfChanges]);
+        }
+      } elseif (isset($oldData[$key]['view'])) {
+        $viewName = $oldData[$key]['view'];
+        $title = $oldData[$key]['title'];
+        // getting the diff of the result of 2 different views
+        $diff = $this->diffViewsData(
+          $newData[$key]['data'],
+          $oldData[$key]['data']
+        );
+        // store the diff only if something has changed
+        if ($diff) {
+          $changes[] = ['view' => $viewName, 'diff' => $diff, 'title' => $title ];
+        }
+      }
+    }
+    $i--;
+  }
+
+  /**
+   * To compare data generated between 2 different moments
+   *
+   * @return ViewsResultDiff
+   */
+  public function compareData() {
+    $this->compareDataRecursive(
+      $this->oldData,
+      $this->newData,
+      $this->changes
+    );
+    return $this;
+  }
+
+  /**
+   * To retrieve the current data of the views configured to diff
+   * this function will be called by storeNewData or storeOldData
+   * to be able to store the data produced by the views for later
+   * comparison
+   *
+   * @return array
+   *     The data retrieved from one of more views,
+   *     joined on specific fields for later comparison
+   */
+  private function getCurrentData($settings = null) {
+    $data = [];
+    $settings = !$settings ? $this->viewsSettings : $settings;
+
+    foreach ($settings as $setting) {
+      // if it is a group then apply recursion
+      if (isset($setting['group'])) {
+        $data[] = ['group' => $this->getCurrentData($setting['group'])];
+      } else if (isset($setting['view'])) {
+        // if it is a view it must be previewed to get the results
+        $view = views_get_view($setting['view']);
+
+        // Setting Exposed Filters
+        // backing up $_GET
+        $_GETbkp = $_GET;
```

Comment:
This seems really messy, is there no other way to get view data without doing this?

Reply:
@mickadoo not currently. I tried using arguments for this but wasn't enough to get the results needed, probably because wrong integration of CiviCRM and views. So I have to use exposed filters, and as far as I know, this is the only way of achieving it, and for sure the simplest.